lib/chef/knife/user_create.rb

```diff
@@ -30,61 +33,72 @@ class UserCreate < Knife
       option :file,
         :short => "-f FILE",
         :long  => "--file FILE",
-        :description => "Write the private key to a file"
+        :description => "Write the private key to a file if the server generated one."
+
+      option :user_key,
+        :long => "--user-key FILENAME",
+        :description =>  "Set the initial default key for the user from a file on disk (cannot pass with --create-key)."
 
-      option :admin,
-        :short => "-a",
-        :long  => "--admin",
-        :description => "Create the user as an admin",
+      option :prevent_keygen,
+        :short => "-k",
+        :long  => "--prevent-keygen",
+        :description => "API V1 only. Prevent server from generating a default key pair for you. Cannot be passed with --user-key.",
         :boolean => true
 
-      option :user_password,
-        :short => "-p PASSWORD",
-        :long => "--password PASSWORD",
-        :description => "Password for newly created user",
-        :default => ""
+      banner "knife user create USERNAME DISPLAY_NAME FIRST_NAME LAST_NAME EMAIL PASSWORD (options)"
 
-      option :user_key,
-        :long => "--user-key FILENAME",
-        :description => "Public key for newly created user.  By default a key will be created for you."
+      def user
+        @user_field ||= Chef::User.new
+      end
 
-      banner "knife user create USER (options)"
+      def create_user_from_hash(hash)
+        Chef::User.from_hash(hash).create
+      end
 
       def run
-        @user_name = @name_args[0]
+        test_mandatory_field(@name_args[0], "username")
+        user.username @name_args[0]
 
-        if @user_name.nil?
-          show_usage
-          ui.fatal("You must specify a user name")
-          exit 1
-        end
+        test_mandatory_field(@name_args[1], "display name")
+        user.display_name @name_args[1]
+
+        test_mandatory_field(@name_args[2], "first name")
```

Comment:
So instead of failing here if the rest are not specified, could we just assume that we need to use osc_user and call that. That way, we don't break backwards compatibility. We could also spit out a deprecation warning of some sort.